src/status_im2/contexts/chat/messages/content/image/view.cljs

```diff
@@ -0,0 +1,34 @@
+(ns status-im2.contexts.chat.messages.content.image.view
+  (:require [react-native.core :as rn]
+            [react-native.fast-image :as fast-image]
+            [utils.re-frame :as rf]))
+
+(defn calculate-dimensions
+  [width height]
+  (let [max-width  (if (> width height) 320 190)
+        max-height (if (> width height) 190 320)]
+    (if (> height width)
+      (let [calculated-height (* (min height max-height) (/ (max width max-width) width))
+            calculated-width  (* (max width max-width) (/ (min height max-height) height))]
+        {:width calculated-width :height calculated-height})
+      (let [calculated-height (* (max height max-height) (/ (min width max-width) width))
+            calculated-width  (* (min width max-width) (/ (max height max-height) height))]
+        {:width calculated-width :height calculated-height}))))
+
+
+(defn image-message
+  [{:keys [content image-width image-height] :as message}]
+  (let [dimensions (calculate-dimensions image-width image-height)]
+    (fn []
+      (let [shared-element-id (rf/sub [:shared-element-id])]
+        [rn/touchable-opacity
+         {:active-opacity 1
+          :on-press       (fn []
+                            (rf/dispatch [:chat.ui/update-shared-element-id (:message-id message)])
```

Comment:
Since you're already destructuring `message` and there's more than one usage of `:message-id`, you could destructure that key as well.